c++/test/TestTimezone.cc

```diff
@@ -437,4 +437,29 @@ namespace orc {
     }
   }
 
+  TEST(TestTimezone, testTzdbFromCondaEnv) {
+    const char* tzDirBackup = std::getenv("TZDIR");
+    // test only makes sense if TZDIR exists
+    if (tzDirBackup != nullptr) {
+      ASSERT_TRUE(delEnv("TZDIR"));
+
+      // remove "/share/zoneinfo" from TZDIR to get (equivalent of) CONDA_PREFIX
+      std::string condaPrefix(tzDirBackup);
+      condaPrefix += "/../..";
+      ASSERT_TRUE(setEnv("CONDA_PREFIX", condaPrefix.c_str()));
```

Comment:
Hm, so I thought I'd avoid the dangling pointer problem here as `condaPrefix` is alive for the duration of the `setEnv` call, but it seems that it doesn't work and produces a garbage path.

Reply:
It's possible that the pointer is just empty because we deleted TZDIR too soon. Let's try reversing that.